### ISSUE TITLE
--cert isn't handled correctly in vagrant box add

### ISSUE THREAD

**Issue by samdbmg** [hashicorp/vagrant#10670]
It looks like the `--cert` option isn't passed through to curl correctly in `vagrant box add` or `vagrant box update`.

### Vagrant version
Vagrant 2.2.3

### Host operating system
macOS High Sierra 10.13.6

### Guest operating system
Ubuntu Bionic

### Vagrantfile
```ruby
# Minimized example
Vagrant.configure("2") do |config|
  # Configure some default vars
  forge_cert_location = ENV.has_key?("FORGE_CERT") ? ENV["FORGE_CERT"] : "~/.certs/devcert.pem"

  # Retrieve custom box
  config.vm.box = "cloudfitbox"
  config.vm.box_url = "https://artifactory.virt.ch.bbc.co.uk/artifactory/api/vagrant/ap-vagrant/cloudfitbox"
  config.vm.box_download_client_cert = forge_cert_location
end
```

### Debug output
https://gist.github.com/samdbmg/874feb7d1071cfcdddaffd3fc6040b32

### Expected behavior
`vagrant box add --cert ~/certs/devcert.pem https://artifactory.virt.ch.bbc.co.uk/artifactory/api/vagrant/ap-vagrant/cloudfitbox` and `vagrant box update` should pass the client certificate to the `curl` command, which `vagrant up` does.

### Actual behavior
The `curl` command doesn't get the `--cert` option added; I think because https://github.com/hashicorp/vagrant/blob/master/plugins/commands/box/command/add.rb#L89 and https://github.com/hashicorp/vagrant/blob/master/plugins/commands/box/command/update.rb#L161 should read `box_download_client_cert: download_options[:client_cert],`

### Steps to reproduce
1. Put a Vagrant box behind a client certificate
2. Try to add or update it with the CLI.
3. 😢 

### References
- GH-4691 seems to have renamed the option to `box_download_client_cert` in the Action
- GH-4473 seems to have regressed the client cert feature

**Comment by samdbmg** [comment]
I'll try and get Vagrant building for me tomorrow (Ruby isn't my first language) and put a PR in if I can.

**Comment by briancain** [comment]
Hey @samdbmg - the debug log you shared shows that you actually just used the command incorrectly. If you're adding a box that way with a URL or file, you need to give it a name:

```
 INFO interface: error: A name is required when adding a box file directly. Please pass
the `--name` parameter to `vagrant box add`. See
`vagrant box add -h` for more help.
A name is required when adding a box file directly. Please pass
the `--name` parameter to `vagrant box add`. See
`vagrant box add -h` for more help.
```

You typed `vagrant box add --cert "/Users/samn/certs/devcert.pem" "https://artifactory.virt.ch.bbc.co.uk/artifactory/api/vagrant/ap-vagrant/cloudfitbox"` but it's missing some required arguments with the `add` command.

Do you have debug output showing Vagrant actually failing when making a request with your cert?

**Comment by samdbmg** [comment]
Good point @briancain - that wasn't a great example. I've updated the gist with the output of: `VAGRANT_LOG=info vagrant box add --name cloudfitbox --cert ~/certs/devcert.pem https://artifactory.virt.ch.bbc.co.uk/artifactory/api/vagrant/ap-vagrant/cloudfitbox &> vagrant-box-add-with-name-info.txt`
See https://gist.github.com/samdbmg/874feb7d1071cfcdddaffd3fc6040b32#file-vagrant-box-add-with-name-info-txt

The interesting bit is lines [159](https://gist.github.com/samdbmg/874feb7d1071cfcdddaffd3fc6040b32#file-vagrant-box-add-with-name-info-txt-L159) and [175](https://gist.github.com/samdbmg/874feb7d1071cfcdddaffd3fc6040b32#file-vagrant-box-add-with-name-info-txt-L175) which show the `curl` command without the `--cert ~/certs/devcert.pem`. Compare to fetching the box when running `vagrant up` ([vagrant-up-info.txt#L243](https://gist.github.com/samdbmg/874feb7d1071cfcdddaffd3fc6040b32#file-vagrant-up-info-txt-L243)) where the cert flag is added correctly.

**Comment by samdbmg** [comment]
Ah, didn't notice when I opened the issue, but #10622 fixes this I think

**Comment by briancain** [comment]
> INFO subprocess: Starting process: ["/opt/vagrant/embedded/bin/curl", "-q", "--fail", "--location", "--max-redirs", "10", "--verbose", "--user-agent", "Vagrant/2.2.3 (+https://www.vagrantup.com; ruby2.4.4) ", "--continue-at", "-", "--cert", "/Users/samn/certs/devcert.pem", "-H", "Accept: application/json", "--output", "/Users/samn/.vagrant.d/tmp/box14d808ae1971b347f7146080b9a5a25e8bef9303", "https://artifactory.virt.ch.bbc.co.uk/artifactory/api/vagrant/ap-vagrant/cloudfitbox"]

Hmm, just to double check, it looks like Vagrant is actually using the right cert here on your `vagrant-up-info.txt` log. I see it as the `--cert` argument to curl when it makes the request to download your box and is successful in doing so in the debug log. Was that with the changes from #10622 ?

**Comment by samdbmg** [comment]
No, all these logs came from 2.2.3

When running `vagrant up` without the box installed, the `--cert` argument is passed to curl from the Vagrantfile (obviously with the box installed, nothing is downloaded!) 

However if you run `vagrant box update` or `vagrant add --cert {cert_path} {box_url} ...` (as above) the `--cert` argument isn't passed to curl.

Using the fork in #10622 solves the problem; `vagrant box update` and `vagrant add ...` both work as expected.

**Comment by briancain** [comment]
Fixed via #10622. Thanks!
